### PR TITLE
Add conf-graphics to check for the Graphics library

### PR DESCRIPTION
**Pull request by lefessan**
archimedes fails to install when Graphics is not available, which is the case of ocaml-3.12.1 compiled with opam on Debian, see:

 http://opam.ocamlpro.com/builder/html/report-last.html#archimedes

I think it is better if the failure is detected earlier using a conf-graphics package.

### CODE REVIEW

File: packages/conf-graphics/conf-graphics.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,8 @@
+opam-version: "1"
+maintainer: "Fabrice Le Fessant <fabrice.le_fessant@ocamlpro.com>"
+homepage: "http://www.ocaml.org/"
+license: "LGPL+EXN-BIN"
+build: [["ocamlc" "-custom" "graphics.cma" "-o" "test"]]
+post-messages: [
+  "This package checks whether the Graphics library was compiled." {failure}
```
Comment (avsm) [inline packages/conf-graphics/conf-graphics.1.0/opam]: some hints in here about how to install it? e.g. the brew install for xquartz here would be a common failure i suspect https://medium.com/partially-applied/stubbed-toes-getting-started-with-ocaml-on-os-x-9ff16768a12d#.bjg7h0wio

Comment (lefessan) [inline packages/conf-graphics/conf-graphics.1.0/opam]: Unfortunately, I have no such hint: I only use Debian, and even on it, the ./configure of OCaml fails to detect X11 dev files on some versions. Anyway, the Graphics library will probably be moved out of OCaml soon, as many of the other libraries, so it is a temporary problem.

**Comment by lefessan** [comment]
Maybe the package should be called base-graphics instead of conf-graphics, as it is provided by the OCaml distribution, for now.

**Comment by dsheets** [comment]
It could be called just `graphics` so dependencies don't change if/when it moves out of the distribution.

**Comment by Chris00** [comment]
Incidentally, I had to solve a related problem recently where I wanted [conditional compilation](https://github.com/Chris00/mesh/blob/master/setup.ml#L41) depending on the presence (or not) of the `Graphics` module.

**Comment by Chris00** [comment]
> Shall I copy "graphics" into `depopts` instead of `depends`, 

Yes — archimedes is supposed to compile without requiring `Graphics`.

> and replace the "--%{ocamlfind:enable}%-graphics" by "--enable-graphics" {"%{graphics:installed}%"} ?

Aren't the two form equivalent?

**Comment by lefessan** [comment]
I don't know the form `%{ocamlfind:enable}%`. For me, it sounds a bit magic that this variable has a link with `graphics`, and not only with `ocamlfind`.

**Comment by samoht** [comment]
> I don't know the form %{ocamlfind:enable}%. For me, it sounds a bit magic that this variable has a link with graphics, and not only with ocamlfind.

I think @Chris00 meant you should use `%{graphics:enable}%--graphics`. See the "Package-local" variable section of https://opam.ocaml.org/doc/manual/dev-manual.html#sec37

**Comment by lefessan** [comment]
That's what my last commit does, the `"--%{ocamlfind:enable}%-graphics"` is what was written in the original `opam` file, it was probably a bug.

**Comment by Chris00** [comment]
You should use `--%{graphics:enable}%--graphics`.  The `ocamlfind` version was the original version because the `ocamlfind` package [says that it provides `graphics`](https://github.com/ocaml/opam-repository/blob/master/packages/ocamlfind/ocamlfind.1.6.2/findlib).  This should be removed with the new `graphics` package (which should contain a `findlib` file).

**Comment by samoht** [comment]
```
#=== ERROR while installing graphics.1.0 ======================================#
# opam-version 1.2.2
# os           darwin
# command      ocamlc -custom graphics.cma -o test
# path         /Users/travis/.opam/system/build/graphics.1.0
# compiler     system (4.02.3)
# exit-code    2
# env-file     /Users/travis/.opam/system/build/graphics.1.0/graphics-6342-865888.env
# stdout-file  /Users/travis/.opam/system/build/graphics.1.0/graphics-6342-865888.out
# stderr-file  /Users/travis/.opam/system/build/graphics.1.0/graphics-6342-865888.err
### stderr ###
# File "_none_", line 1:
# Error: Cannot find file graphics.cma
```

Can you also update https://github.com/ocaml/ocaml-ci-scripts/blob/master/.travis-ocaml.sh#L94-L99 to fix that? Thanks!

**Comment by lefessan** [comment]
@samoht I don't have OSX here, so I cannot update the Travis script, as I don't know what to change in it. Anyway, this PR is just moving an already existing error from a package (archimedes) to another one (graphics) to make it easier to understand the failure, so I don't think a change is needed anywhere else.

**Comment by lefessan** [comment]
I merge since I cannot fix the Darwin problem (that did exist prior to this PR).

**Pull request merged**